### DISCUSSION TITLE
Set-WallPaper: The term 'Set-WallPaper' is not recognized as a name of a cmdlet, function, script file, or executable program.

### DISCUSSION THREAD

**Discussion by rogue-agent** [PowerShell/PowerShell#16597] [Q&A]
Hello,

I'm working on QIP (Quick Install Package) which would automatically install programs and change the wallpaper but well.

I have never been able to change the wallpaper because of the issue:
>` Set-WallPaper -Image ".\bin\other\wp.png"`

>`Set-WallPaper: The term 'Set-WallPaper' is not recognized as a name of a cmdlet, function, script file, or executable program.`

How am I supposed change the wallpaper, either CMD or with PS?

**Comment by vexx32** [comment]
There isn't a way in native PowerShell, you have to use p/invoke which means compiling some C# to handle the interface. It's doable, though. Here's a function that will do the thing you want:


https://gist.github.com/vexx32/e8128cbc19881fb965a286cf0bc613fc

(code cleaned up and reworked a bit from what I found here: https://www.joseespitia.com/2017/09/15/set-wallpaper-powershell-function/)

**Reply by rogue-agent** [reply]
If this is not a native function then how is PS going to replace CMD as the defacto Terminal interface for NT platform?
